python/tests/table/test_metadata.py

```diff
@@ -0,0 +1,279 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import io
+import json
+from uuid import UUID
+
+import pytest
+
+from iceberg.schema import Schema
+from iceberg.serializers import FromByteStream
+from iceberg.table.metadata import TableMetadata, TableMetadataV1, TableMetadataV2
+from iceberg.types import NestedField, StringType
+
+EXAMPLE_TABLE_METADATA_V1 = {
+    "format-version": 1,
+    "table-uuid": UUID("aefee669-d568-4f9c-b732-3c0cfd3bc7b0"),
+    "location": "s3://foo/bar/baz.metadata.json",
+    "last-updated-ms": 1600000000000,
+    "last-column-id": 4,
+    "schema": {
+        "schema-id": 0,
+        "fields": [
+            {"id": 1, "name": "foo", "required": True, "type": "string"},
+            {"id": 2, "name": "bar", "required": True, "type": "string"},
+            {"id": 3, "name": "baz", "required": True, "type": "string"},
+            {"id": 4, "name": "qux", "required": True, "type": "string"},
+        ],
+        "identifier-field-ids": [],
+    },
+    "schemas": [
+        {
+            "schema-id": 0,
+            "fields": [
+                {"id": 1, "name": "foo", "required": True, "type": "string"},
+                {"id": 2, "name": "bar", "required": True, "type": "string"},
+                {"id": 3, "name": "baz", "required": True, "type": "string"},
+                {"id": 4, "name": "qux", "required": True, "type": "string"},
+            ],
+            "identifier-field-ids": [],
+        },
+    ],
+    "current-schema-id": 0,
+    "partition-spec": {},
```

Comment:
In v1, this should be a list. Initially, this stored only the partition fields because we didn't support multiple partition specs. See https://iceberg.apache.org/spec/#table-metadata

Reply:
I wasn't aware of that, thanks!